Mathlib/NumberTheory/LSeries/HurwitzZetaEven.lean

```diff
@@ -0,0 +1,386 @@
+  ext1 x
+  simp only [← QuotientAddGroup.mk_zero, ← ofReal_inj, evenKernel_def, ofReal_zero, sq, mul_zero,
+    zero_mul, Complex.exp_zero, one_mul, cosKernel_def]
+
+lemma evenKernel_neg (a : UnitAddCircle) (x : ℝ) : evenKernel (-a) x = evenKernel a x := by
+  induction' a using QuotientAddGroup.induction_on' with a'
+  simp only [← QuotientAddGroup.mk_neg, ← ofReal_inj, evenKernel_def, ofReal_neg, neg_sq, neg_mul,
+    jacobiTheta₂_neg_left]
+
+lemma cosKernel_neg (a : UnitAddCircle) (x : ℝ) : cosKernel (-a) x = cosKernel a x := by
+  induction' a using QuotientAddGroup.induction_on' with a'
+  simp only [← QuotientAddGroup.mk_neg, ← ofReal_inj, cosKernel_def, ofReal_neg,
+    jacobiTheta₂_neg_left]
+
+lemma continuousOn_evenKernel (a : UnitAddCircle) : ContinuousOn (evenKernel a) (Ioi 0) := by
+  induction' a using QuotientAddGroup.induction_on' with a'
+  apply continuous_re.comp_continuousOn (f := fun x ↦ (evenKernel a' x : ℂ))
+  simp only [evenKernel_def a']
+  refine ContinuousAt.continuousOn (fun x hx ↦ ((Continuous.continuousAt ?_).mul ?_))
+  · exact Complex.continuous_exp.comp (continuous_const.mul continuous_ofReal)
+  · have h := continuousAt_jacobiTheta₂ (a' * I * x) (?_ : 0 < im (I * x))
+    · exact h.comp (f := fun u : ℝ ↦ (a' * I * u, I * u)) (by fun_prop)
+    · rwa [mul_im, I_re, I_im, zero_mul, one_mul, zero_add, ofReal_re]
+
+lemma continuousOn_cosKernel (a : UnitAddCircle) : ContinuousOn (cosKernel a) (Ioi 0) := by
+  induction' a using QuotientAddGroup.induction_on' with a'
+  apply continuous_re.comp_continuousOn (f := fun x ↦ (cosKernel a' x : ℂ))
+  simp only [cosKernel_def]
+  refine ContinuousAt.continuousOn (fun x hx ↦ ?_)
+  have : 0 < im (I * x) := by rwa [mul_im, I_re, I_im, zero_mul, one_mul, zero_add, ofReal_re]
+  exact (continuousAt_jacobiTheta₂ a' this).comp (f := fun u : ℝ ↦ (_, I * u)) (by fun_prop)
+
+lemma evenKernel_functional_equation (a : UnitAddCircle) (x : ℝ) :
+    evenKernel a x = 1 / x ^ (1 / 2 : ℝ) * cosKernel a (1 / x) := by
+  rcases le_or_lt x 0 with hx | hx
+  · rw [evenKernel_undef _ hx, cosKernel_undef, mul_zero]
+    exact div_nonpos_of_nonneg_of_nonpos zero_le_one hx
+  induction' a using QuotientAddGroup.induction_on' with a
+  rw [← ofReal_inj, ofReal_mul, evenKernel_def, cosKernel_def, jacobiTheta₂_functional_equation]
+  have h1 : I * ↑(1 / x) = -1 / (I * x) := by
+    push_cast
+    rw [← div_div, mul_one_div, div_I, neg_one_mul, neg_neg]
+  have h2 : (a * I * x / (I * x)) = a := by
+    field_simp [hx.ne']
+    ring -- `rw [mul_assoc]` works, but we don't want to rely on the exact output of `field_simp`
+  have h3 : 1 / (-I * (I * x)) ^ (1 / 2 : ℂ) = 1 / ↑(x ^ (1 / 2 : ℝ)) := by
+    rw [neg_mul, ← mul_assoc, I_mul_I, neg_one_mul, neg_neg,ofReal_cpow hx.le, ofReal_div,
+      ofReal_one, ofReal_ofNat]
+  have h4 : -π * I * (a * I * x) ^ 2 / (I * x) = - (-π * a ^ 2 * x) := by
+    rw [mul_pow, mul_pow, I_sq]; field_simp [hx.ne']; ring_nf
```

Comment:
```suggestion
  have h4 : -π * I * (a * I * x) ^ 2 / (I * x) = - (-π * a ^ 2 * x) := by
    rw [mul_pow, mul_pow, I_sq, div_eq_iff hx']
    ring
```
Together with the previous one makes the proof twice as fast. (` field_simp`  is slow here.)